receiver/datadogreceiver/internal/translator/traces_translator.go

```diff
@@ -90,6 +90,13 @@ func ToTraces(payload *pb.TracerPayload, req *http.Request) ptrace.Traces {
 	// now instead of being a span level attribute.
 	groupByService := make(map[string]ptrace.SpanSlice)
 
+	// Datadog traces split a 128 bits trace id in two parts: TraceID and Tags._dd_p_tid. This happens if the
+	// instrumented service received a TraceContext from an OTel instrumented service. When it happens, we need
+	// to concatenate the two into newSpan.TraceID.
+	// The following map keeps track of the TraceIDs we process as only the first span has the upper 64 bits from the
+	// 128 bits trace ID.
+	map64to128 := make(map[uint64]pcommon.TraceID)
```

Comment:
I'm concerned about an unbounded cache. While each entry is small, this is just unbounded memory growth.

It looks like the `spanmetricsconnector` uses a LRU (least recently used) cache from hashicorp `"github.com/hashicorp/golang-lru/v2/simplelru"` that would be a good candidate to use here to ensure there is a limit to the size of this cache.

Reply:
This is a good point, the cache would just be bound to that one `ToTraces` run, but I'm not sure what a good size for it would be. Should it be configurable? If not, what would a good value be?

Reply:
Actually it seems fairly easy to add such a config, I'll add that.

Reply:
Done in [90eb40c](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/39654/commits/90eb40ca0f6cbf2cb734dfbe863e78f0b97a6387)